cpp/src/fil/fil.cu

```diff
@@ -328,7 +330,7 @@ struct forest {
   int fixed_block_count_ = 0;
   int max_shm_           = 0;
   // Optionally used
-  rmm::device_uvector<float> vector_leaf_;
+  rmm::device_uvector<char> vector_leaf_;
```

Comment:
Do we have `byte` defined somewhere? If yes, it's better to use it here. If no, then `std::uint8_t` would do. Either of them is better than `char`.

Reply:
Done (with `std::uint8_t`).